### PR TITLE
Community hub updates

### PR DESCRIPTION
**Pull request by minimalsm**
## Description

[Deploy preview of the new community hub page](https://deploy-preview-4521--ethereumorg.netlify.app/en/community/)

## Related Issue

This is an epic of our Q4 roadmap: #4215

### CODE REVIEW

File: src/components/Nav/index.js [hunk 1]
```diff
@@ -323,7 +323,8 @@ const Nav = ({ handleThemeChange, isDarkTheme, path }) => {
     }
   }
 
-  const shouldShowSubNav = path.includes("/developers/")
+  const shouldShowSubNav =
```
Comment (samajammin) [inline src/components/Nav/index.js:326]: Is the thinking here to have a community subnav on the community pages?

I'm open to this but would probably default to just adding these subpages to the Community dropdown nav.

File: src/content/community/online/index.md [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+- [r/ethtrader](https://www.reddit.com/r/ethtrader/) - _trends & market analysis_
+- [r/EtherMining](https://www.reddit.com/r/EtherMining/) - _focused on securing the Ethereum Network (mining)_
+- [r/ethstaker](https://www.reddit.com/r/ethstaker/) - _welcome to all interested in staking on Ethereum_
+- [Fellowship of Ethereum Magicians](https://ethereum-magicians.org) - _community oriented around technical standards in Ethereum_
+- [Ethereum Stackexchange](https://ethereum.stackexchange.com) - _discussion and help for Ethereum developers_
+- [Ethereum Research](https://ethresear.ch) - _the most influential messageboard for cryptoeconomic research_
+
+## Chat rooms
+
+- <Icon name="discord" /> <a href="https://gitter.im/ethereum/home">Ethereum Gitter</a> - <i>chat room for the Ethereum GitHub repo</i>
```
Comment (samajammin) [inline src/content/community/online/index.md:28]: I'm not sure we should link out to these Gitter channels anymore. Most of the ones I checked out seem outdated, unmoderated & full of scams.

File: src/pages/community.js [hunk 3]
```diff
@@ -0,0 +1,683 @@
...
+  // }
+
+  const heroContent = {
+    title: "Join the community",
+    header: "Welcome to the Ethereum community hub",
+    subtitle:
+      "The Ethereum community is home to hundreds of thousands of developers, technologists, designers, users, HODLers and enthusiasts.",
+    image: data.enterprise.childImageSharp.fluid,
+    alt: translateMessageId("page-community-hero-alt", intl),
+    buttons: [
```
Comment (samajammin) [inline src/pages/community.js:336]: What should our primary CTAs be here? 🤔 

Should we have any? Or let users scroll down to see options?

File: src/pages/community.js [hunk 4]
```diff
@@ -0,0 +1,683 @@
...
+                How can I get involved?
+              </H2>
+              <Subtitle>
+                {/* <Translation id="page-community-get-involved-description" /> */}
+                There are many ways to get involved in the fast-growing Ethereum
+                community; you can join one of the popular online communities,
+                attend an event, join a meetup group, contribute to a project,
+                or participate in one of the many online forums about Ethereum.
+              </Subtitle>
+              <ButtonLink to="/community/grants">
```
Comment (samajammin) [inline src/pages/community.js:451]: Another option question - do we want a primary CTA here? 

Perhaps a primary button here to "get involved" & a secondary to "explore online communities"?

Or perhaps we let the user scroll to see the options below?

**Comment by samajammin** [comment]
One general comment on the homepage "get involved" section:
Image 2021-11-24 at 6 49 11 PM

I think all these make sense _except_ for "get support" doesn't really strike me as a way to "get involved".

I might re-structure this to have 3 main choices/cards under "get involved":
- get involved
- events
- online communities

Then I'd probably have a separate section like the ones below for "find support".

**Comment by samajammin** [comment]
Just my take, I'd be curious to hear what others think - feels a bit weird to list the meetups on the homepage but not list upcoming events. I feel like maybe we just display meetups on the events page.

File: src/pages/community.js [hunk 5]
```diff
@@ -0,0 +1,683 @@
...
+      title: "Join an online community",
+      description:
+        "Find your tribe and participate in community with other Ethereum enthusiasts",
+      alt: translateMessageId("page-index-get-started-wallet-image-alt", intl),
+      to: "/community/online/",
+    },
+    {
+      image: data.eth.childImageSharp.fixed,
+      title: "Find Ethereum events",
+      description: "We've identified impactful work that needs doing.",
```
Comment (samajammin) [inline src/pages/community.js:361]: This description feels off - I'd go with something around "Browse conferences, hackathons and meetups around the world you can attend"

File: src/pages/community.js [hunk 6]
```diff
@@ -0,0 +1,683 @@
...
+            {/* <StyledH2>
+              <Translation id="page-index-what-is-ethereum" />
+            </StyledH2> */}
+            <H2>
+              {/* <Translation id="page-community-open-source" /> */}
+              Creator? Builder? Get paid for your work.
+            </H2>
+            <Subtitle>
+              {/* <Translation id="page-community-open-source-description" /> */}
+              Are you building on Ethereum, or do you want to? Funding grants
```
Comment (samajammin) [inline src/pages/community.js:492]: I like this section - maybe also mentions there's companies hiring for thousands of jobs? i.e. make it more broad about working in the space vs. just grants?

File: src/pages/community.js [hunk 7]
```diff
@@ -0,0 +1,683 @@
...
+              are available to help get your project off the ground, and these
+              projects are looking for passionate people to work for them..
+            </Subtitle>
+            <ButtonRow>
```
Comment (samajammin) [inline src/pages/community.js:496]: I like these 2 options of 1) explore grants & 2) find a job.

File: src/pages/community.js [hunk 8]
```diff
@@ -0,0 +1,683 @@
...
+      <SupportContainer>
+        <RowReverse>
+          <FeatureContent>
+            <H2>
+              {/* <Translation id="page-community-support" /> */}
+              Developer support
+            </H2>
+            <Subtitle>
+              {/* <Translation id="page-community-support-description" /> */}
+              The Ethereum community has adopted many standards that are helpful
```
Comment (samajammin) [inline src/pages/community.js:568]: This feels very specific 🤔  - I think a dedicated section to general "Ethereum support" would be the way to go vs. getting into e.g. EIPs or specifically support for devs. I think we can cover that on the support page.

Comment (minimalsm) [reply src/pages/community.js:568]: This was placeholder from the original designs. I'm not actually sure what we should put here (or if we want a 3rd secondary call to action). I've left it in for now just incase to save writing it twice.

**Comment by Aldekein** [comment]
On https://deploy-preview-4521--ethereumorg.netlify.app/en/community/ in the very end of the page I see unparsed Markdown code:

image

File: src/components/MeetupList.js [hunk 9]
```diff
@@ -310,23 +310,42 @@ const LeftContainer = styled.div`
...
-        </RightContainer>
-      </Item>
-    ))}
-  </Table>
-)
+const MeetupList = () => {
+  const [searchField, setSearchField] = useState("")
+  const filteredMeetups = filterMeetups(searchField)
+
+  const handleSearch = (event) => setSearchField(event.target.value)
```
Comment (corwintines) [inline src/components/MeetupList.js:328]: I know we have talked about this, but would add the location to the search here so we filter by the name of the event, as well as the city/location for the event.

File: src/intl/en/common.json [hunk 10]
```diff
@@ -62,12 +62,15 @@
   "ethereum-2-0": "Ethereum 2.0",
   "ethereum-brand-assets": "Ethereum Brand Assets",
   "ethereum-community": "Ethereum Community",
+  "ethereum-online": "Online communities",
```
Comment (corwintines) [inline src/intl/en/common.json:65]: More of a general note, but we have some mixed casing in the Community drop down now with these. There is a bit of this throughout all of the drop downs, but we should probably look at addressing that.

File: src/content/community/get-involved/index.md [hunk 11]
```diff
@@ -0,0 +1,111 @@
...
+
+Join thousands of node operators in helping to further decentralize Ethereum.
+
+- [More on how to run a node](/developers/docs/nodes-and-clients/run-a-node/)
+
+### Stake your ETH {#staking}
+
+By staking your ETH you can earn rewards whilst helping to secure the Ethereum network.
+
+-[More on staking](/eth2/staking/)
```
Comment (corwintines) [inline src/content/community/get-involved/index.md:62]:
```suggestion
- [More on staking](/eth2/staking/)
```
Just adding a space for markdown

File: src/content/community/get-involved/index.md [hunk 12]
```diff
@@ -0,0 +1,111 @@
...
+
+- Learn about and try Ethereum at [ethereum.org/developers/](/developers/)
+- [Find a bounty on Gitcoin](https://gitcoin.co/), work on a small or large technical issue, earn crypto!
+- Attend an [ETHGlobal](http://ethglobal.co/) hackathon near you!
+- Check out [projects related to your area of expertise or programming language of choice](/developers/docs/programming-languages/)
+- Watch or participate in the [Core Dev calls](https://www.youtube.com/playlist?list=PLaM7G4Llrb7zfMXCZVEXEABT8OSnd4-7w)
+- [Ecosystem Support Program's wishlist](https://esp.ethereum.foundation/wishlist/) - tooling, documentation, and infrastructure areas where the Ethereum Ecosystem Support Program is actively seeking grant applications
+- [Web3Bridge](https://www.web3bridge.com/) - join the aspiring web3 community in their initiative to identify, train, and support hundreds of developers and community members throughout Africa
+
+## Researchers & Academics <Emoji text=":mag:" size={1} />‍ {#researchers-and-academics}
```
Comment (corwintines) [inline src/content/community/get-involved/index.md:22]: These headers with emojis seem to have a weird side effect on the side panel 
Screen Shot 2021-12-06 at 2 46 26 PM

File: src/content/community/online/index.md [hunk 13]
```diff
@@ -0,0 +1,44 @@
...
+- [Ethereum Foundation]() - _Keep upto date with the latest from the Ethereum Foundation_
+- [@ethereum]() - _Offical account of the Ethereum Foundation_
+- [@ethdotorg]() - _The portal to Ethereum, built for our growing global community_
+- [List of influential Ethereum twitter accounts]()
+
+<Divider />
+
+## Explore other online Ethereum communities
+
+- [DAOs](/community/daos/)
```
Comment (corwintines) [inline src/content/community/online/index.md:44]: Is this going to end up at `http://localhost:8000/en/community/get-involved/#decentralized-autonomous-organizations-daos`?

File: src/content/community/online/index.md [hunk 14]
```diff
@@ -0,0 +1,44 @@
...
+- <Icon name="discord" /> <a href="https://discord.gg/5W5tVb3">CryptoDevs</a> - <i>Ethereum development focused Discord community</i>
+- <Icon name="discord" /> <a href="https://discord.io/ethstaker">EthStaker Discord</a> - <i>community oriented around offering project management support to Ethereum development</i>
+- <Icon name="discord" /> <a href="https://discord.gg/CetY6Y4">Ethereum.org website team</a> - <i>stop by and chat ethereum.org web development and design with the team and folks from the community</i>
+
+## YouTube and Twitter
+
+- [Ethereum Foundation]() - _Keep upto date with the latest from the Ethereum Foundation_
+- [@ethereum]() - _Offical account of the Ethereum Foundation_
+- [@ethdotorg]() - _The portal to Ethereum, built for our growing global community_
+- [List of influential Ethereum twitter accounts]()
```
Comment (corwintines) [inline src/content/community/online/index.md:38]: Do we have a list of these twitter accounts somewhere?

File: src/content/community/support/index.md [hunk 15]
```diff
@@ -0,0 +1,77 @@
...
+### Execution clients {#execution-clients}
+
+- [Geth](https://discord.gg/FqDzupGyYf)
+- [Nethermind](https://discord.gg/YJx3pm8z5C)
+- [Besu](https://discord.gg/p8djYngzKN)
+- [Ergion](https://github.com/ledgerwatch/erigon/issues) <!-- Does Erigon have a public facing channel? -->
+
+### Consensus clients {#consensus-clients}
+
+- [Prysm](https://discord.gg/prysmaticlabs)
```
Comment (corwintines) [inline src/content/community/support/index.md:69]: In the spirit of diversity, is there a way to randomize this? Probably not in markdown, just curious.

File: src/content/community/support/index.md [hunk 16]
```diff
@@ -0,0 +1,77 @@
...
+- [Nimbus](https://discord.gg/nSmEH3qgFv)
+- [Lighthouse](https://discord.gg/cyAszAh)
+- [Teku](https://discord.gg/7hPv2T6)
+- [Lodestar](https://discord.gg/aMxzVcr)
+
+## Common Questions
+
+- How do I....?
```
Comment (corwintines) [inline src/content/community/support/index.md:77]: Do you want a hand wtih some of these?

File: src/pages/community.js [hunk 17]
```diff
@@ -0,0 +1,676 @@
...
+          </FeatureContent>
+          <ImageContainer>
+            <FeatureImage
+              fluid={data.hackathon.childImageSharp.fluid}
+              alt={translateMessageId("page-community-support-alt", intl)}
+            />
+          </ImageContainer>
+        </RowReverse>
+      </SupportContainer>
+      <Content>
```
Comment (corwintines) [inline src/pages/community.js:582]: To me I think this is redundant when we have a dedicated events page, and link out to it above this section.

File: src/content/community/events/index.md [hunk 18]
```diff
@@ -0,0 +1,24 @@
...
+description: How to get involved in the Ethereum community.
+sidebar: false
+lang: en
+---
+
+## Upcoming events {#events}
+
+**Every month, there are major Ethereum events around the world.** Consider attending one near you to meet more people in the community, learn about employment opportunities, and develop new skills.
+
+<UpcomingEventsList/>
```
Comment (corwintines) [inline src/content/community/events/index.md:12]: Wasn't noticing the UpcomingEventsList rendering. I think we talked about this already though, but if you need a hand let me know.

File: src/pages/community.js [hunk 19]
```diff
@@ -0,0 +1,676 @@
...
+`
+
+const SupportContainer = styled.div`
+  background: ${(props) => props.theme.colors.homeBoxPurple};
+  display: flex;
+  align-items: center;
+  flex-direction: row;
+  height: 720px;
+  width: 100%;
+  margin-top: -1px;
```
Comment (corwintines) [inline src/pages/community.js:255]: This container seems a little crowded on the top in mobile
Screen Shot 2021-12-06 at 3 08 44 PM

File: src/pages/community.js (hunk 4 above)
Comment (corwintines) [reply src/pages/community.js:445]: I think just scrolling here is good

**Comment by corwintines** [comment]
Great start so far @minimalsm! Let me know if you have any questions from above, or if you want a hand :)

File: src/content/community/get-involved/index.md (hunk 12 above)
Comment (minimalsm) [reply src/content/community/get-involved/index.md:22]: [Fixed in this commit](https://github.com/ethereum/ethereum-org-website/pull/4521/commits/ea7a1c7b6f03f08677f6c7795a772b1cf1aa1825) but doesn't seem very elegant if anyone wants to refactor :-)

File: src/content/community/events/index.md (hunk 18 above)
Comment (minimalsm) [reply src/content/community/events/index.md:12]: We ran out of events!

- [Added events here](https://github.com/ethereum/ethereum-org-website/pull/4521/commits/1ceb4a31d71e8a1fa50a738f0db34931d820e00e)
- [Starter code for individual cards for each event](https://github.com/ethereum/ethereum-org-website/pull/4521/commits/dd9423be95871778336616a65ec828fe20ace945)

Comment (minimalsm) [reply src/content/community/events/index.md:12]: Amongst other things would love if we could add some styling to connect cards akin to [how it is done here](https://bitcoin.org/en/events)

File: src/content/community/support/index.md (hunk 15 above)
Comment (minimalsm) [reply src/content/community/support/index.md:69]: Mmm not sure how we would do that. But if possible to keep the content in markdown but randomize somehow I say lets do it :-)

File: src/pages/community.js (hunk 17 above)
Comment (minimalsm) [reply src/pages/community.js:612]: [Removed](https://github.com/ethereum/ethereum-org-website/pull/4521/commits/4d77177c2234e509e0a908d53ae0b5e9ac80afb7) and replaced it with some [call outs to other pages](https://github.com/ethereum/ethereum-org-website/pull/4521/commits/c564ab28c60dec35d01b2f1193535f84d02924f9)

File: src/content/community/support/index.md (hunk 16 above)
Comment (minimalsm) [reply src/content/community/support/index.md]: [Added some initial questions.](https://github.com/ethereum/ethereum-org-website/pull/4521/commits/b0d947a23a7f2ed1e4f5344c3b4318a575189ca1) Feel free to suggest any others. I think with ~5ish questions we can get an 80/20 here without it being information overload.

File: src/content/community/online/index.md (hunk 14 above)
Comment (minimalsm) [reply src/content/community/online/index.md]: [Added](https://github.com/ethereum/ethereum-org-website/pull/4521/commits/47ac70deba8b6942fbbbea9e9681a011c7c32ce1)

File: src/content/community/events/index.md (hunk 18 above)
Comment (samajammin) [reply src/content/community/events/index.md:12]: We should have some fallback if there's 0 upcoming events, e.g. "We're not aware of any upcoming events. Know of one? Please add it!" with a link to create an issue.

File: src/components/Icon.js [hunk 20]
```diff
@@ -58,8 +87,10 @@ Icon.propTypes = {
 }
 
 const StyledIcon = styled(Icon)`
-  fill: ${(props) => props.theme.colors.secondary};
-  &:hover path:last-of-type {
+  fill: ${(props) =>
+    props.color ? props.color : props.theme.colors.secondary};
+
+  &:hover path {
```
Comment (samajammin) [inline src/components/Icon.js:93]: I think this change is creating some weird behavior on the nav dropdown icons:

https://user-images.githubusercontent.com/8097623/145498288-95bbeb52-7c33-4454-99b9-2cfc7cb5e809.mp4

Comment (wackerow) [reply src/components/Icon.js]: Yeah, still seeing this as well

File: src/components/UpcomingEventsList.js [hunk 21]
```diff
@@ -35,30 +51,50 @@ const UpcomingEventsList = () => {
...
   }, [])
 
+  if (orderedUpcomingEvents?.length === 0) {
+    return (
+      <InfoBanner emoji=":information_source:">
+        {" "}
+        We're not aware of any upcoming events. Know of one?{" "}
+        <Link to="https://github.com/ethereum/ethereum-org-website/blob/dev/src/data/community-events.json">
+          Please add it to this page!
+        </Link>
```
Comment (wackerow) [inline src/components/UpcomingEventsList.js:75]: Again here, should we tee this up for translation support?

File: src/content/community/support/index.md (hunk 15 above)
Comment (wackerow) [reply src/content/community/support/index.md:73]: Would consider listing Prysm lower solely for the sake of subtly promoting client diversity. We should try to emphasize the other client communities.

File: src/components/Icon.js (hunk 20 above)
Comment (wackerow) [reply src/components/Icon.js]: Just pushed a patch for this

**Review by wackerow — APPROVED** [review]
This is looking fantastic! Just took a look through, patched one bug when hovering over the menu icons.

Noticed we haven't set up the `MeetupList` or `UpcomingEventsList` for `intl` support... were we planning on getting these translated?

Also left a quick comment about not listing Prysm first, but slapping an approval on here since we can always update the above in a separate PR in favor of getting this shipped 💪🏻

**Pull request merged**